packages/react-reconciler/src/ReactFiberUnwindWork.js

```diff
@@ -215,12 +217,20 @@ function throwException(
               : renderExpirationTime;
 
           // Attach a listener to the promise to "ping" the root and retry.
-          const onResolveOrReject = retrySuspendedRoot.bind(
+          let onResolveOrReject = retrySuspendedRoot.bind(
             null,
             root,
             workInProgress,
             pingTime,
           );
+          if (enableInteractionTracking) {
+            // Wrap suspense callback so that the current interaction are preserved.
+            // Interaction threads are unique per root and expiration time.
+            // DO NOT CHANGE THIS VALUE without also updating other threadID values.
+            const threadID =
+              renderExpirationTime * 1000 + root.interactionThreadID;
+            onResolveOrReject = wrap(onResolveOrReject, threadID);
```

Comment:
This is not obvious to me. Wouldn’t there already be an interaction tracked for this expiration time? And these callbacks can’t do arbitrary things. They just schedule work on something that is already scheduled. So seems to me that this shouldn’t need to be wrapped.

@acdlite

Reply:
Yes, we're within an active interaction zone at this point. (That's why `wrap` works in the first place.)

The issue is that we attach `onResolveOrReject` to our thennable to resume work when the e.g. promise has finished loading. `wrap` ensures that we restore the right set of interactions at that time.

Reply:
> Without this, our interaction zone expires when React finishes this batch of work– and there's nothing to restore it when the thenable completes.

This is only an issue if the work times out, which we determine here: https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberScheduler.js#L1449

Reply:
Sorry, maybe I'm being dense, but I'm not sure I understand how this feedback applies.